core/src/main/java/org/infinispan/partitionhandling/impl/PartitionHandlingManager.java

```diff
@@ -20,5 +28,87 @@
    CacheTopology getLastStableTopology();
+
+   /**
+    * Adds a partially aborted transaction.
+    * <p/>
+    * The transaction should be registered when it is not sure if the abort happens successfully in all the affected
+    * nodes. It can rethrow the exception is the transaction is not registered.
+    *
+    * @param globalTransaction the global transaction.
+    * @param affectedNodes     the nodes involved in the transaction and they must abort the transaction.
+    * @param lockedKeys        the keys locally locked.
+    * @param suspectException  the suspected exception. It contains the node who left the view.
+    */
+   void addPartialRollbackTransaction(GlobalTransaction globalTransaction, Collection<Address> affectedNodes,
+                                      Collection<Object> lockedKeys, SuspectException suspectException);
+
+   /**
+    * Adds a partially committed transaction.
+    * <p/>
+    * The transaction is committed in the second phase and it is register if it is not sure that the transaction was
+    * committed successfully in all the affected nodes. It can rethrow the exception if the transaction is not
+    * registered.
+    *
+    * @param globalTransaction the global transaction.
+    * @param affectedNodes     the nodes involved in the transaction and they must commit it.
+    * @param lockedKeys        the keys locally locked.
+    * @param newVersions       the updated versions. Only used when versioning is enabled.
+    * @param suspectException  the suspected exception. It contains the node who left the view.
+    */
+   void addPartialCommit2PCTransaction(GlobalTransaction globalTransaction, Collection<Address> affectedNodes,
+                                       Collection<Object> lockedKeys, EntryVersionsMap newVersions, SuspectException suspectException);
+
+   /**
+    * Adds a partially committed transaction.
+    * <p/>
+    * The transaction is committed in one phase and it is register if it is not sure that the transaction was committed
+    * successfully in all the affected nodes. It can rethrow the exception if the transaction is not registered.
+    *
+    * @param globalTransaction the global transaction.
+    * @param affectedNodes     the nodes involved in the transaction and they must commit it.
+    * @param lockedKeys        the keys locally locked.
+    * @param modifications     the transaction's modification log.
+    * @param suspectException  the suspected exception. It contains the node who left the view.
+    */
+   void addPartialCommit1PCTransaction(GlobalTransaction globalTransaction, Collection<Address> affectedNodes,
+                                       Collection<Object> lockedKeys, List<WriteCommand> modifications, SuspectException suspectException);
+
+   /**
+    * It checks if the transaction resources (for example locks) can be released.
+    * <p/>
+    * The transaction resource can't be released when the transaction is partially committed.
```

Comment:
The name doesn't seem to fit the PHM's job profile very well, how about renaming it to something like `isTransactionPartiallyCommitted()`?